### DISCUSSION TITLE
Why use Exception as the Result error type instead of Error or ErrorException?

### DISCUSSION THREAD

**Discussion by carlosjortiz** [louthy/language-ext#1287] [Ideas]
I was reviewing the `Result`, `Error`, and `ErrorException` codes and noticed that you are not using these error types as the `Result` error type. I would like to understand the reason behind this, and if it aligns with your preferences, I am willing to make the necessary adjustments

**Comment by louthy** [comment]
Result is an intermediate type for Try (and Try variants) and not designed for direct consumption; it will be deprecated due to the endless confusion about it.  Fin is the type you want, it uses Error for its 'alternative' state.

**Reply by carlosjortiz** [reply]
Thk
